transports/jms/src/main/java/org/mule/transport/jms/JmsMessageDispatcher.java

```diff
@@ -185,17 +191,17 @@ private MuleMessage dispatchMessage(MuleEvent event, boolean doSend, final Compl
                     {
                         if (topic)
                         {
-                            return internalBlockingSendAndAwait(consumer, producer, replyTo, jmsMessage, topic, ttl, priority, persistent, timeout);
+                            return internalBlockingSendAndAwait(session, producer, replyTo, jmsMessage, topic, ttl, priority, persistent, timeout, event);
 
                         }
                         else
                         {
-                            return internalBlockingSendAndReceive(producer, consumer, replyTo, jmsMessage, topic, ttl, priority, persistent, timeout);
+                            return internalBlockingSendAndReceive(session, producer, replyTo, jmsMessage, topic, ttl, priority, persistent, timeout, event);
                         }
                     }
                     finally
                     {
-                        closeConsumer(session, consumer, replyTo);
+                        closeReplyQueue(session, replyTo);
```

Comment:
Did you review code to ensure everything, consumer, reply queue, destination etc etc are all be closing correctly in each of the 3 or 4 scenarios after this change?